bin/bootstrax

```diff
@@ -1453,16 +1477,32 @@ def process_run(rd, send_heartbeats=args.production):
                 send_heartbeat(update)
             ec = strax_proc.exitcode
             if ec is None:
+                # fail(bla) raises RunFailed so no need for elifs. Make sure to
+                # kill main process before raising errors in fail(bla)
+
+                # Fail because we are taking a very long time to process
                 if t0 < now(-timeouts['max_processing_time']):
+                    kill_process(strax_proc.pid)
                     fail(f"Processing took longer than {timeouts['max_processing_time']} sec")
+
+                # Fail because we are taking a very long time to process despite the run having ended
+                if (run_coll.find_one({'_id': rd['_id']}, projection={'end': 1}).get('end', None) is not None
```

Comment:
This feels like it should be a helper function. `get_end_time(rd['_id'])` or something